tests/study_tests/test_study.py

```diff
@@ -799,9 +799,21 @@ def _objective(trial: Trial) -> float:
 def test_optimize_with_progbar_parallel(capsys: _pytest.capture.CaptureFixture) -> None:
 
     study = create_study()
-    with pytest.warns(UserWarning, match="Progress bar only supports serial execution"):
-        study.optimize(lambda _: 1.0, n_trials=10, show_progress_bar=True, n_jobs=-1)
+    study.optimize(lambda _: 1.0, n_trials=10, show_progress_bar=True, n_jobs=2)
+    _, err = capsys.readouterr()
+
+    # Search for progress bar elements in stderr.
+    assert "10/10" in err
+    assert "100%" in err
```

Comment:
I found that most of the lines in this test function are duplicated with `test_optimize_with_progbar`. If we apply `parametrize` to `test_optimize_with_progbar` in terms of `n_jobs`, we can reduce the code and improve the maintainability.

```python
@pytest.mark.parametrize("n_jobs", [1, 2])
def test_optimize_with_progbar(n_jobs: int, capsys: _pytest.capture.CaptureFixture) -> None:

    study = create_study()
    study.optimize(lambda _: 1.0, n_trials=10, show_progress_bar=True, n_jobs=n_jobs)
    _, err = capsys.readouterr()

    # Search for progress bar elements in stderr.
    assert "10/10" in err
    assert "100%" in err
```

In addition, I also found some missing test cases, for example, (`n_jobs=2`, `n_trials=10`, `timeout=2`) and (`n_jobs=2`, `n_jobs=None`, `timeout=None`). So, please apply `parametrize` to the following test functions as well.

- `test_optimize_without_progbar`
- `test_optimize_progbar_n_trials_prioritized`
- `test_optimize_progbar_no_constraints`

Reply:
> I found that most of the lines in this test function are duplicated with test_optimize_with_progbar. If we apply parametrize to test_optimize_with_progbar in terms of n_jobs, we can reduce the code and improve the maintainability.

Ok, fixed.

> In addition, I also found some missing test cases, for example, (n_jobs=2, n_trials=10, timeout=2) and (n_jobs=2, n_jobs=None, timeout=None). So, please apply parametrize to the following test functions as well.

Ok, fixed.

| show_progress_bar | n_trials | timeout | n_jobs | covered by |
| ------------- | ------------- | ------------- | ------------- | ------------- |
| False | None | None | 1 | test_optimize_without_progbar_no_constraints |
| False | None | None | not 1 | test_optimize_without_progbar_no_constraints |
| False | None | not None | 1 | test_optimize_without_progbar_timeout |
| False | None | not None | not 1 | test_optimize_without_progbar_timeout |
| False | not None | None | 1 | test_optimize_without_progbar |
| False | not None | None | not 1 | test_optimize_without_progbar |
| False | not None | not None | 1 | test_optimize_without_progbar_n_trials_prioritized |
| False | not None | not None | not 1 | test_optimize_without_progbar_n_trials_prioritized |
| True | None | None | 1 | test_optimize_progbar_no_constraints |
| True | None | None | not 1 | test_optimize_progbar_no_constraints |
| True | None | not None | 1 | test_optimize_with_progbar_timeout |
| True | None | not None | not 1 | test_optimize_with_progbar_parallel_timeout |
| True | not None | None | 1 | test_optimize_with_progbar |
| True | not None | None | not 1 | test_optimize_with_progbar |
| True | not None | not None | 1 | test_optimize_progbar_n_trials_prioritized |
| True | not None | not None | not 1 | test_optimize_progbar_n_trials_prioritized |